### ISSUE TITLE
Build fail with New Architecture

### ISSUE THREAD

**Issue by cervisebas** [react-native-community/react-native-share#1415]
### Steps to reproduce
1. yarn add react-native-share
2. Active New Architecture
3. npx react-native run-android 

### Expected behaviour
Build success and install debug app

### Actual behaviour
Build fail:
```
C:\Projects\Test\node_modules\react-native-share\android\src\newarch\cl\json\RNShare.java:16: error: RNShare is not abstract and does not override abstract method getTypedExportedConstants() in NativeRNShareSpec
public class RNShare extends NativeRNShareSpec {
       ^
C:\Projects\AsistenciaDigital\node_modules\react-native-share\android\src\newarch\cl\json\RNShare.java:32: error: getConstants() in RNShare cannot override getConstants() in NativeRNShareSpec
    public Map<String, Object> getConstants() {
                               ^
  overridden method is final
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
2 errors

FAILURE: Build failed with an exception.
```

### Environment
- **React Native version**: 0.72.1
- **React Native platform + platform version**: Android 11
- **Typescript version**: 4.8.4

### react-native-share
**Version**: 9.2.0

**Comment by ErHarinderSingh** [comment]
@cervisebas same here

**Comment by ErHarinderSingh** [comment]
iOS fix is here.
https://github.com/sayem314/react-native-keep-awake/pull/9
**Fix podspec for RN<=0.70 and new arch enabled**

**Comment by cervisebas** [comment]
Temporarily fix this error by making the following changes:
- In the file "RNShare.java" in line 32 change:
```java
@Override
public Map<String, Object> getConstants() {
    return delegate.getConstants();
}
```
by the following:
```java
@Override
public Map<String, Object> getTypedExportedConstants() {
    return delegate.getConstants();
}
```

**Comment by MateusAndrade** [comment]
> iOS fix is here. [sayem314/react-native-keep-awake#9](https://github.com/sayem314/react-native-keep-awake/pull/9) **Fix podspec for RN<=0.70 and new arch enabled**

It looks like we already have this fix: https://github.com/react-native-share/react-native-share/blob/c495549cd399c7da15e6bd067ccf4475995a62dc/RNShare.podspec#L5

> Temporarily fix this error by making the following changes:
> 
> * In the file "newarch/cl/json/RNShare.java" in line 32 change:
> 
> ```java
> @Override
> public Map<String, Object> getConstants() {
>     return delegate.getConstants();
> }
> ```
> 
> by the following:
> 
> ```java
> @Override
> public Map<String, Object> getTypedExportedConstants() {
>     return delegate.getConstants();
> }
> ```

Niceee! Would you mind opening a PR with that? 😃

**Comment by cervisebas** [comment]
> > iOS fix is here. [sayem314/react-native-keep-awake#9](https://github.com/sayem314/react-native-keep-awake/pull/9) **Fix podspec for RN<=0.70 and new arch enabled**
> 
> It looks like we already have this fix: https://github.com/react-native-share/react-native-share/blob/c495549cd399c7da15e6bd067ccf4475995a62dc/RNShare.podspec#L5
> 
> > Temporarily fix this error by making the following changes:
> > 
> > * In the file "newarch/cl/json/RNShare.java" in line 32 change:
> > 
> > ```java
> > @Override
> > public Map<String, Object> getConstants() {
> >     return delegate.getConstants();
> > }
> > ```
> > 
> > by the following:
> > 
> > ```java
> > @Override
> > public Map<String, Object> getTypedExportedConstants() {
> >     return delegate.getConstants();
> > }
> > ```
> 
> Niceee! Would you mind opening a PR with that? 😃 
> 

Done, I just created a PR
